### PR TITLE
Fix locale determination in confirm/choice inputs

### LINKED ISSUE #4981: PORT: Fix locale determination in confirm/choice prompts/inputs
Port https://github.com/microsoft/botbuilder-js/pull/2534 to .NET

### PR DESCRIPTION
**Pull request by v-kydela**
Fixes #4981

## Description

This is a port of https://github.com/microsoft/botbuilder-js/pull/2534, and it should help adaptive dialogs handle unexpected locales better in Composer

## Specific Changes

- Implement `PromptCultureModelsTests.ShouldNotThrowWhenLocaleIsNull`
- Fix `MapToNearestLanguage` so that it doesn't throw when the locale is null and so it's more efficient and easier to read and closer to the TypeScript implementation
- Rename `GetCulture` to `DetermineCulture` in `ConfirmInput` and `ChoiceInput` to align with other classes
- Fix `DetermineCulture` in `ConfirmInput` and `ChoiceInput` so that it uses `MapToNearestLanguage`
- Have `DetermineCulture` accept a second parameter in `ChoiceInput` to align with `ChoicePrompt`
- Rework `ConfirmPrompt` and `ChoicePrompt` constructors so that the default choice options are only generated when needed, aligning with best practices and the TypeScript implementations
- Clean up `ConfirmPrompt`, `ChoicePrompt`, `ConfirmInput`, and `ChoiceInput` according to Visual Studio suggestions

## Testing
`PromptCultureModelsTests.ShouldNotThrowWhenLocaleIsNull`

### CODE REVIEW

**Review by ceciliaavila — APPROVED** [review]

**Comment by v-kydela** [comment]
@microsoft/bf-adaptive 
@microsoft/botframework-sdk 

Please review

File: libraries/Microsoft.Bot.Builder.Dialogs/Prompts/PromptCultureModels.cs [hunk 1]
```diff
@@ -235,28 +235,27 @@ public static class PromptCultureModels
...
-                                cultureCode.StartsWith(o.Split('-').First(), StringComparison.Ordinal)).ToList();
+                cultureCode = cultureCode.ToLowerInvariant();
 
-                if (fallbackCultureCodes.Count == 1)
+                if (!SupportedLocales.Contains(cultureCode))
                 {
-                    return fallbackCultureCodes.First();
-                }
+                    var culturePrefix = cultureCode.Split('-').First();
+                    var fallbackLocales = SupportedLocales.Where(locale => locale.StartsWith(culturePrefix, StringComparison.Ordinal));
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Builder.Dialogs/Prompts/PromptCultureModels.cs:246]: Multiple enumerations of fallbackLocales in the lines that follow. Consider converting to List.

**Comment by mrivera-ms** [comment]
>         /// ChoiceSet or expression which evalutes to a ChoiceSet.

Typo. Please fix.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:71 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

**Comment by mrivera-ms** [comment]
>         /// ChoiceOptions or expression which evluates to ChoiceOptions.

Typo. Please fix.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:107 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

**Comment by mrivera-ms** [comment]
>         public ObjectExpression<FindChoicesOptions> RecognizerOptions { get; set; } = null;

Not needed.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:119 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

**Comment by mrivera-ms** [comment]
>         /// Replaces the result with the FoundChoice value if possible, then proceedes to <see cref="Dialog.ResumeDialogAsync(DialogContext, DialogReason, object, CancellationToken)"/>.

Typo. Please fix.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:122 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs [hunk 2]
```diff
@@ -127,10 +128,9 @@ public ChoiceInput([CallerFilePath] string callerPath = "", [CallerLineNumber] i
...
         /// or threads to receive notice of cancellation.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        public override Task<DialogTurnResult> ResumeDialogAsync(DialogContext dc, DialogReason reason, object result = null, CancellationToken cancellationToken = default(CancellationToken))
+        public override Task<DialogTurnResult> ResumeDialogAsync(DialogContext dc, DialogReason reason, object result = null, CancellationToken cancellationToken = default)
         {
-            FoundChoice foundChoice = result as FoundChoice;
-            if (foundChoice != null)
+            if (result is FoundChoice foundChoice)
             {
                 // return value insted of FoundChoice object
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:135]: >insted [](start = 32, length = 6)

Typo. Please fix.

**Comment by mrivera-ms** [comment]
>         /// Called when input has been received, recognices choice.

Typo. Please fix.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:171 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

**Comment by mrivera-ms** [comment]
>         /// The locale (en-us, nl-nl, etc) or expression which evluates to locale.

Typo. Please fix.

---
Refers to: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ConfirmInput.cs:56 in 54c40ed. [](commit_id = 54c40edfe92b5525d0702748f0228547acd5ceae, deletion_comment = False)

**Comment by v-kydela** [comment]
@mrivera-ms - I have applied your fixes

**Review by mrivera-ms — APPROVED** [review]
:shipit:

**Pull request merged**